analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierBuilder.cs

```diff
@@ -58,11 +68,34 @@ private VerifierBuilder(VerifierBuilder original)
             new(this) { Paths = Paths.Concat(paths).ToImmutableArray() };
 
         public VerifierBuilder AddReferences(IEnumerable<MetadataReference> references) =>
-            new(this) { References = References.Concat(references).ToImmutableArray() };
+            references != null && references.Any()
```

Comment:
we evaluate `references` enumerable multiple times.

Reply:
I gave it a try, but it's not very nice. Should I just remove the `Any` check instead?

Reply:
Yes. Does it bring any value :) (pun intended)

Reply:
We can actually simplify it even more, as the null could come only from the `OldVerifier` anyway and should not happen under normal usage.